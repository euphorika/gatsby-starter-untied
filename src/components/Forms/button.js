import React from 'react'

import styles from './button.module.styl'

const Button = ({ children, type, options }) => (
  <button className={styles.button} type={type} {...options}>
    {children}
  </button>
)

export default Button
