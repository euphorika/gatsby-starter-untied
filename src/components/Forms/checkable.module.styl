@require '../../themes/standard/components/checkable.styl'

:root
  --checkmark-color: checkable.checkmark-color
  --background-color: checkable.background
  --hover-background-color: checkable.hover-background
  --checked-background-color: checkable.checked-background
  --width: checkable.dimensions
  --height: checkable.dimensions

.label
  position relative

  &:hover
    .element
      & ~ .checkmark
        background var(--hover-background-color)

  .element
    position absolute
    opacity 0
    cursor pointer
    height 0
    width 0

    &:checked
      & ~ .checkmark
        background var(--checked-background-color)

        &::after
          display block

  .checkmark
    position absolute
    top 0
    left 0
    height var(--height)
    width var(--width)
    background-color var(--background-color)

    &::after
      content ""
      position absolute
      display none
