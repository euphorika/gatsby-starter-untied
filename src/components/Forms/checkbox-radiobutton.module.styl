.label
  position relative

  &:hover
    .element
      & ~ .checkmark
        background #ccc

  .element
    position absolute
    opacity 0
    cursor pointer
    height 0
    width 0

    &:checked
      & ~ .checkmark
        background #2196F3

        &::after
          display block

  .checkmark
    position absolute
    top 0
    left 0
    height 25px
    width 25px
    background-color #eee

    &::after
      content ""
      position absolute
      display none
