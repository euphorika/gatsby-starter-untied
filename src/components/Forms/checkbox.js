import React from 'react'
import PropTypes from 'prop-types'

import styles from './checkbox.module.styl'

const Checkbox = ({ value, checked, name, label, options }) => {
  const checkbox = (
    <>
      <input
        className={`${styles.element} ${styles.autoWidth}`}
        type="checkbox"
        value={value}
        name={name}
        checked={checked}
        {...options}
      />
      <span className={styles.afterFormElement} />
    </>
  )

  if (label) {
    return (
      <label className={styles.label}>
        {checkbox}
        {label}
      </label>
    )
  }

  return checkbox
}

export default Checkbox

Checkbox.propTypes = {
  value: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  options: PropTypes.object
}
