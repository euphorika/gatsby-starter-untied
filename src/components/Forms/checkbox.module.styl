@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

@require './element.module.styl'
@require './checkbox-radiobutton.module.styl'

.label
  .checkmark
    &::after
      left 9px
      top 5px
      width 5px
      height 10px
      border solid white
      border-width 0 3px 3px 0
      transform rotate(45deg)
