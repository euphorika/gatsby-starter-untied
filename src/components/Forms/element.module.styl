@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.label
  padding 10px

  .element
    margin-left 10px

.element
  padding 10px
  width 100%

  &.auto-width
    width auto

    &:required
      width auto

  &:required
    width 95%

    & + .after-form-element
      &::before
        content '*'
        display inline-block
        width 5%
        color red

  &:valid
    border-color: green

  &:invalid
    border-color: red
