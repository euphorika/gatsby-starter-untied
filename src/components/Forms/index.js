import React from 'react'

import styles from './styles.module.styl'

const Forms = () => <section className={styles.formsSection}>Forms</section>

export default Forms
