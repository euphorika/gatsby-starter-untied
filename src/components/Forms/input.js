import React from 'react'

import styles from './input.module.styl'

const Input = ({ value, name, type = 'text', label, options }) => {
  const input = (
    <>
      <input
        className={styles.element}
        {...options}
        value={value}
        name={name}
        type={type}
      />
      <span className={styles.afterInput} />
    </>
  )

  if (label) {
    return (
      <label>
        {label}
        {input}
      </label>
    )
  }

  return input
}

export default Input
