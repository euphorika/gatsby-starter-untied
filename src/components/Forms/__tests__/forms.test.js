import React from 'React'
import renderer from 'react-test-renderer'
import Forms from '../'

describe('Forms', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Forms />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
