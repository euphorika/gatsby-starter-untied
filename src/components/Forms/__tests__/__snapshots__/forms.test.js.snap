// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forms renders correctly 1`] = `
<section
  className="formsSection"
>
  Forms
</section>
`;
